task/host-sp-comms/src/main.rs

```diff
@@ -847,11 +866,30 @@ impl ServerImpl {
                 if response_len == 0 {
                     self.tx_buf.reset();
                     return Err(KeyLookupResult::NoValueForKey);
-                } else if response_len > max_response_len {
-                    self.tx_buf.reset();
-                    return Err(KeyLookupResult::MaxResponseLenTooShort);
                 }
+                response_len
             }
+            Key::InventorySize => {
+                // We reply with a tuple of count, API version:
+                const REPLY: (u32, u32) =
+                    (ServerImpl::INVENTORY_COUNT, INVENTORY_API_VERSION);
+                self.tx_buf.encode_response(
+                    sequence,
+                    &SpToHost::KeyLookupResult(KeyLookupResult::Ok),
+                    |buf| {
+                        const_assert!(
+                            MIN_SP_TO_HOST_FILL_DATA_LEN
+                                >= core::mem::size_of::<u32>() * 2
+                        );
+                        hubpack::serialize(buf, &REPLY).unwrap_lite()
+                    },
+                );
+                core::mem::size_of_val(&REPLY)
```

Comment:
This is correct, but looks a little weird at first glance: we're using hubpack to serialize, but then returning `mem::size_of_val` for the number of bytes serialized. Those are definitely the same, but could we add an assertion on the value returned by `hubpack::serialize()` to document that?

Reply:
Yeah, I switched to matching existing behavior in f5f02fc2da82d467b4cd48baf051344a109e246c